### PR TITLE
Implement basic timer popover for dashboard and task views

### PR DESCRIPTION
**Pull request by jrbotros**

### CODE REVIEW

File: orchestra/views.py [hunk 1]
```diff
@@ -258,8 +258,26 @@ def get_timer(request):
     worker = Worker.objects.get(user=request.user)
     try:
         if request.method == 'GET':
-            duration = time_tracking.get_timer_current_duration(worker)
-            return str(duration)
+            timer = time_tracking.get_timer_object(worker)
+            time_worked = time_tracking.get_timer_current_duration(worker)
+            data = TaskTimerSerializer(timer).data
+            if time_worked:
+                data['time_worked'] = str(time_worked)
```
Comment (jrbotros) [inline orchestra/views.py:265]: Maybe we should just add this to the serializer?

File: orchestra/migrations/0041_tasktimer_description.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (lydiagu) [inline orchestra/migrations/0041_tasktimer_description.py]: fyi i just landed a diff with migration 0041

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    .module('orchestra.timing')
+    .directive('workTimer', function() {
+      return {
+        restrict: 'E',
+        scope: {
+          taskId: '=taskId',
+        },
+        bindToController: true,
+        controllerAs: 'workTimer',
+        templateUrl: '/static/orchestra/timing/timer/timer.html',
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js]: templateUrls need to be wrapped in `$static()` now.

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        controller: function($http, $location, $scope, $timeout, timeEntries) {
+          var workTimer = this;
+
+          workTimer.popoverTemplate = '/static/orchestra/timing/timer/popover.html';
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js]: wrap in `$static` - @joshblum static file processing still works when the template url is set to a variable right?

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            }
+          };
+
+          workTimer.startTimer = function() {
+            var startTimerUrl = '/orchestra/api/interface/timer/start/';
+            $http.post(startTimerUrl, {
+              // TODO(jrbotros): add current assignment here
+            })
+            .then(function(response) {
+              console.log(response.data);
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js]: remove

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+              alert('Could not start timer');
+            });
+          };
+
+          workTimer.stopTimer = function() {
+            var stopTimerUrl = '/orchestra/api/interface/timer/stop/';
+            $timeout.cancel(workTimer.timerPromise);
+            $http.post(stopTimerUrl)
+            .then(function(response) {
+              console.log(response.data);
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js]: remove

File: orchestra/views.py (hunk 1 above)
Comment (lydiagu) [inline orchestra/views.py:265]: yes probably. I can do that later.

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+          workTimer.popoverTemplate = '/static/orchestra/timing/timer/popover.html';
+
+          var resetTimer = function() {
+            workTimer.timerRunning = false;
+            workTimer.description = undefined;
+            workTimer.timeElapsed = moment.duration();
+          };
+
+          workTimer.toggleTimer = function() {
+            if (workTimer.toggledOn) {
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js:28]: why do you need both toggledOn and timerRunning?

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+            });
+          };
+
+          workTimer.stopTimer = function() {
+            var stopTimerUrl = '/orchestra/api/interface/timer/stop/';
+            $timeout.cancel(workTimer.timerPromise);
+            $http.post(stopTimerUrl)
+            .then(function(response) {
+              console.log(response.data);
+              workTimer.toggledOn = false;
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js]: do you need to also set timerRunning to false?

File: orchestra/templates/orchestra/index.html [hunk 9]
```diff
@@ -12,6 +12,7 @@
 </div>
 <!--  notification end -->
 
+<work-timer task-id="1" style="position: absolute; top: 10px; right: 200px;"></work-timer>
```
Comment (lydiagu) [inline orchestra/templates/orchestra/index.html:16]: hard-coded task-id? Don't forget to update later!

**Comment by lydiagu** [comment]
Looking good! Thanks for breaking this into smaller PRs. :)

Done reviewing!

File: orchestra/views.py (hunk 1 above)
Comment (jrbotros) [inline orchestra/views.py:265]: Sounds good ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 3 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 4 above)
Comment (joshblum) [inline orchestra/static/orchestra/timing/timer/directives.js]: yup shouldn't be a problem

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 7 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js:28]: To decouple the display (the toggle button to turn the timer on and off) from the state of the timer—we don't want to immediately say the timer isn't running anymore before we find out from the server if that's actually true.

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 8 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: It's in `resetTimer`, where this line should be as well!

Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: ✅

File: orchestra/templates/orchestra/index.html (hunk 9 above)
Comment (jrbotros) [inline orchestra/templates/orchestra/index.html:16]: Yep, once I implement the task stuff!

Comment (jrbotros) [inline orchestra/templates/orchestra/index.html:16]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 7 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js:28]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 6 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 5 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 4 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: ✅

**Pull request merged**